### PR TITLE
chore: Introduce convenience methods for intstr.

### PR DESCRIPTION
**Pull request by alexec**
Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [x] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [x] I've signed the CLA.
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My builds are green. Try syncing with master if they are not. 
* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo/blob/master/USERS.md).

# Changes

1. Introduce `util/intstr` to add convince methods to.
2. Print out test errors in Github Actions annotation format - so errors appear on source code.

image

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -319,6 +319,10 @@ $(GOPATH)/bin/go-junit-report:
 test-results/junit.xml: $(GOPATH)/bin/go-junit-report test-results/test.out
 	cat test-results/test.out | go-junit-report > test-results/junit.xml
 
+.PHONY: printtestresultannotations
+printtestresultannotations:
```
Comment (simster7) [inline Makefile:323]:
```suggestion
print-test-result-annotations:
```
?

File: hack/print_test_error_annotations.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+	Name      string     `xml:"name,attr"`
+	TestCases []testcase `xml:"testcase"`
+}
+
+type report struct {
+	XMLName    xml.Name    `xml:"testsuites"`
+	TestSuites []testsuite `xml:"testsuite"`
+}
+
+func printTestResultAnnotations() {
```
Comment (simster7) [inline hack/print_test_error_annotations.go:28]: Excellent!

File: util/intstr/parse.go [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package intstr
+
+import "k8s.io/apimachinery/pkg/util/intstr"
+
+// convenience func to get a pointer
+func Parse(val string) *intstr.IntOrString {
```
Comment (simster7) [inline util/intstr/parse.go:6]:
```suggestion
func ParsePointer(val string) *intstr.IntOrString {
```
?

File: Makefile (hunk 1 above)
Comment (alexec) [reply Makefile:323]: this is super hard to type, I will change to `test-report` (like Maven has `mvn test:report`

File: util/intstr/parse.go (hunk 3 above)
Comment (alexec) [reply util/intstr/parse.go:6]: I've made it `ParsePtr` as this is common

**Pull request merged**